pkg/annotations/middleware.go

```diff
@@ -7,55 +7,51 @@ import (
 	"github.com/containous/traefik/v2/pkg/config/dynamic"
 )
 
-type middlewareBuilder func(middleware *dynamic.Middleware, annotations map[string]string) (bool, error)
+type middlewareBuilder func(annotations map[string]string) (*dynamic.Middleware, string, error)
```

Comment:
Please use named returns to make it clearer what this func type will return from reading its prototype